File: exp_data/平衡实验数据处理/plot_fig.m
% 读取数据
filename = '.\3\0.monitor.csv';
data = readtable(filename);

% 根据 t 升序排序
data = sortrows(data, 't');

% 提取 r 值
r_values = data.r;

% 获取数据长度
n = length(r_values);

% 设置采样间隔
interval = 14; % 您可以调整此值以选择采样间隔

% 采样数据
sampled_indices = 1:interval:n;
sampled_r = r_values(sampled_indices);

% 生成横坐标
x_values = sampled_indices;

% 绘制折线图
figure('Position', [100, 100, 1200, 400]); % 调整图像宽度
plot(x_values, sampled_r, '-', 'LineWidth', 1.1);
xlabel('回合数/轮', 'FontSize', 20);
ylabel('奖励值', 'FontSize', 20);
% title('Line Plot of r Values with Sampling');
% grid on;

% 保存图像为文件（可选）
% saveas(gcf, 'line_plot_r_values.bmp');
print('myfigure','-dsvg');  % 导出为 SVG

File: exp_data/平衡实验数据处理/plot_fig.py
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib import rcParams

rcParams['font.sans-serif'] = ['SimHei']  # 使用黑体 ('SimHei') 或其他支持中文的字体
rcParams['axes.unicode_minus'] = False  # 解决负号 '-' 显示为方块的问题

filename = "./ZBicycleBalanceEnv-v0_4/total.csv"
data = pd.read_csv(filename)

data = data.sort_values(by='t')  # 根据 't' 列升序排序
r_values = data['r']  # 提取 'r' 值
n = len(r_values)  # 获取数据长度
interval = 14  # 设置采样间隔

# 采样数据
sampled_indices = range(0, n, interval)
sampled_r = r_values.iloc[sampled_indices]
x_values = list(sampled_indices)  # 生成横坐标

# 绘制折线图
# plt.figure(figsize=(13, 6))
plt.plot(x_values, sampled_r, '-', linewidth=1.1)
plt.xlabel('回合数/轮', fontsize=15)
plt.ylabel('奖励值', fontsize=15)
# plt.title('Line Plot of r Values with Sampling')
# plt.grid()

# plt.savefig('line_plot_r_values.bmp')
plt.savefig('line.svg', format='svg')

# 显示图像
plt.show()
